src/TgSharp.Tests/TgSharpTests.cs

```diff
@@ -243,7 +244,7 @@ public virtual async Task DownloadFileFromContactTest()
                 .FirstOrDefault(x => x.Phone == NumberToSendMessage);
 
             var inputPeer = new TLInputPeerUser() { UserId = user.Id };
-            var res = await client.SendRequestAsync<TLMessagesSlice>(new TLRequestGetHistory() { Peer = inputPeer });
+            var res = (TLMessagesSlice)await client.SendRequestAsync(new TLRequestGetHistory() { Peer = inputPeer });
```

Comment:
why make this worse? why not generic?

Reply:
It's a bit complex to explain, but the gist of it is that SendRequestAsync didn't know what was the response type of the TLRequestGetHistory before so it had to get the response in generic, now it knows BUT the response type of TLRequestGetHistory is TLAbsMessages (an abstraction) which then should be casted accordingly